controllers/openstackdataplanenodeset_controller.go

```diff
@@ -310,35 +319,54 @@ func (r *OpenStackDataPlaneNodeSetReconciler) Reconcile(ctx context.Context, req
 	return ctrl.Result{}, nil
 }
 
-func checkDeployment(helper *helper.Helper,
-	request ctrl.Request,
+func checkDeployment(
+	ctx context.Context,
+	helper *helper.Helper,
+	instance *dataplanev1.OpenStackDataPlaneNodeSet,
 ) (bool, bool, error) {
+
 	// Get all completed deployments
 	deployments := &dataplanev1.OpenStackDataPlaneDeploymentList{}
 	opts := []client.ListOption{
-		client.InNamespace(request.NamespacedName.Namespace),
+		client.InNamespace(instance.Namespace),
 	}
-	err := helper.GetClient().List(context.Background(), deployments, opts...)
+	err := helper.GetClient().List(ctx, deployments, opts...)
 	if err != nil {
 		helper.GetLogger().Error(err, "Unable to retrieve OpenStackDataPlaneDeployment CRs %v")
 		return false, false, err
 	}
 
 	isDeploymentReady := false
 	deploymentExists := false
+
+	// Sort deployments from oldest to newest
+	slices.SortFunc(deployments.Items, func(a, b dataplanev1.OpenStackDataPlaneDeployment) int {
+		if a.CreationTimestamp.String() < b.CreationTimestamp.String() {
```

Comment:
Though we don't expect two deployments running at the same time, CreationTimestamp is when the CRD is created (not fully reconciled i.e services deployed).

Reply:
Good point. I've changed this to check LastTransitionTime on the DeploymentReadyCondition. Likely still not perfect, but I think it's slightly better.